doc/yaml_config.md

```diff
@@ -0,0 +1,54 @@
+YAML Config
+-----------
+
+Why YAML
+========
+
+With our previous experience we discard at the start XML due to its poor readability.
+JSON format we also consider has serious disadvantage - lack of comments.
+So we decide to go with YAML which is still comfortable to read and also machine read.
+Only disadvantage we found is lack of mature schema support for YAML.
+
+Rules for Config
+================
+
+1. No default values in code. All values have to be defined in yaml config.
```

Comment:
That's fine, but we should not crash when something is missing. At least, we should try to display a proper error message.